rust/protocol/src/state/session.rs

```diff
@@ -134,6 +134,16 @@ impl SessionState {
         Ok(self.session.sender_chain.is_some())
     }
 
+    pub(crate) fn get_all_receiver_chains_and_counters(&self) -> Result<Vec<(Vec<u8>, u32)>> {
+        let mut results = vec![];
+        for chain in self.session.receiver_chains.iter() {
+            if let Some(ref chain_key) = chain.chain_key {
+                results.push((chain_key.key.clone(), chain_key.index));
```

Comment:
@ehrenkret-signal Please confirm this is what you meant by ratchet key / current counter in the Google doc

Reply:
chain_key does not sound like the right variable...

There should be one in the session state called ratchet key that should be a public key.

Reply:
Yeah that seemed to me what we wanted here except there is no counter associated with that key
https://github.com/signalapp/libsignal-client/blob/master/rust/protocol/src/proto/storage.proto#L11
and the doc mentions the "current counter" associated with the receiver chain so I'm not sure what counter that would mean.

Reply:
[Discussed with Ehren via Signal] what the doc meant by "ratchet key + current counter" was "sender ratchet public key" + "current chain counter". Doc updated.